erc7824-docs/docs/guides/migration-guide.md

```diff
@@ -12,6 +12,419 @@ import TabItem from '@theme/TabItem';
+
+- **Allocate amount semantics**: The resize operation uses `allocate_amount` where negative values withdraw from the channel to unified balance, and positive values deposit to the channel.
+
+- **Legacy channel migration**: Users with existing channels containing non-zero amounts must either resize them to zero or close them to enable full protocol functionality. The recommended approach is to close old channels and create new ones, because it may be simpler.
+
+### Nitrolite SDK
+
+You should definitely read this section if you are using the Nitrolite SDK.
+
+#### Session Key Configuration
+
+Implementing the new session key protocol changes:
+
+<Tabs>
+  <TabItem value="application" label="Application Session Key">
+
+  ```typescript
+  const authRequest = {
+    address: '0x...',
+    session_key: '0x...',
+    application: 'My Trading App', // Application name for scoped access
+    allowances: [
+      { asset: 'usdc', amount: '1000.0' },
+      { asset: 'eth', amount: '0.5' }
+    ],
+    scope: 'app.create',
+    expires_at: BigInt(Math.floor(Date.now() / 1000) + 7 * 24 * 60 * 60) // 7 days
+  };
+  ```
+
+  </TabItem>
+  <TabItem value="root" label="Root Access (Clearnode)">
+
+  ```typescript
+  const authRequest = {
+    address: '0x...',
+    session_key: '0x...',
+    application: 'clearnode', // Special value for root access
+    allowances: [], // Not enforced for root access
+    scope: 'app.create',
+    expires_at: BigInt(Math.floor(Date.now() / 1000) + 365 * 24 * 60 * 60) // Long expiration recommended
+  };
+  ```
+
+  </TabItem>
+</Tabs>
+
+**Important considerations:**
+- Root access keys (application: "clearnode") cannot perform channel operations after expiration
+- Plan expiration times based on your operational needs
+- Application-scoped keys track cumulative spending against allowances
+
+#### Channel Creation: Zero Initial Deposit Requirement
+
+Channels must now be created with zero initial deposit and funded separately via the `resizeChannel` method:
+
+<Tabs>
+  <TabItem value="before" label="Before">
+
+  ```typescript
+  const { channelId } = await client.createChannel({
+    chain_id: 1,
+    token: tokenAddress,
+    amount: BigInt(1000000), // Initial deposit
+    session_key: '0x...' // Optional
+  });
+  ```
+
+  </TabItem>
+  <TabItem value="after" label="After">
+
+  ```typescript
+  // Step 1: Create channel with zero deposit
+  const { channelId } = await client.createChannel({
+    chain_id: 1,
+    token: tokenAddress,
+    // amount and session_key parameters removed
+  });
+  
+  // Step 2: Fund the channel separately
+  await client.resizeChannel({
+    channel_id: channelId,
+    amount: BigInt(1000000),
+  });
+  ```
+
+  </TabItem>
+</Tabs>
+
+#### Resize Operations with Channel Balance Rules
+
+Implementing the new resize semantics with `resize_amount` and `allocate_amount`:
```

Comment:
Here we should add that the resize should be done with `resize_amount = -allocate_amount` not to put channels into misconfigured state and so that Clearnode does not reject ops.